pip/qsharp/_qsharp.py

```diff
@@ -349,6 +349,27 @@ def __str__(self) -> str:
     def _repr_html_(self) -> str:
         return self.__data._repr_html_()
 
+    def check_eq(self, state: Dict[int, complex]) -> bool:
+        """
+        Checks if the state dump is equal to the given state. This is not mathematical equality,
+        as the check ignores global phase.
+
+        :param state: The state to check against, provided as a dictionary of state indices to complex amplitudes.
+        """
+        phase = None
+        # Filter out zero states
+        state = {k: v for k, v in state.items() if v != 0}
+        if len(state) != len(self):
+            return False
+        for key in state:
+            if key not in self.__inner:
+                return False
+            if phase is None:
+                phase = self.__inner[key] / state[key]
+            elif phase != self.__inner[key] / state[key]:
```

Comment:
Do we want to do comparisons here exactly, or with some tolerance? I ran into having to do them with tolerance pretty quickly, since real tests compare not hardcoded numbers but results of simulation which is not 100% precise

Reply:
I adjusted the comparison to use a tolerance parameter that defaults to 1e-10 (a common value I've seen us use before). That should make it easy for the common case and configurable when needed. Updated the tests to verify tolerance behaves as expected.